_layouts/single.html

```diff
@@ -42,6 +42,9 @@
             </nav>
           </aside>
         {% endif %}
+        {% if site.show_edit_on_github_link %}
+        <p><a href="{% github_edit_link %}">{{ site.edit_on_github_link_text }}</a></p>
```

Comment:
I think you should consider building the link like this instead: 

    https://github.com/{{site.repository}}/edit/gh-pages/{{ page.path }}

Then github is not needed as a dependency and it will still work.

Reply:
@Isolesen - don't think `https://github.com/{{site.repository}}/edit/gh-pages/{{ page.path }}` will work as what happens when you don't use the `gh-pages` brand? The path will vary.

By not needing GitHub as a dependency... are you referring to GitHub Pages? Not sure if that's a big deal. I suppose `{% github_edit_link %}` could be replaced with a `site.` variable that you could define the base path. But that adds another layer of complexity and something else for the user to customize.

I really don't want to overthink this as I wasn't completely sold on the feature needing to be in the core theme. Only really considered it since `{% github_edit_link %}` made things easier to generate the link.